### ISSUE TITLE
Allow configuration of proxy for connections between Zeebe client and cluster

### ISSUE THREAD

**Issue by Aafi01** [camunda/camunda#9851]
**Is your feature request related to a problem? Please describe.**
I am using Camunda 8 SaaS as my Zeebe cluster. My Java client application is behind company proxy. Client connection to the cloud cluster keeps failing due to the presence of a forward proxy. I couldn't find any documentation for configuring proxy in Zeebe Java client.

**Describe the solution you'd like**
I want an option to specify proxy details (host, port and authentication credentials) to the Zeebe client builder. This configuration should only apply to the client-cluster communication and should not affect other HTTP calls made in the client application.

**Describe alternatives you've considered**
I have tested the client-cluster connection to be working by enabling proxy at the system level. This is not ideal scenario as it also affects other applications running on that system. Also, the client application makes REST calls to intranet applications which shouldn't go through the proxy. I could add those URLs to ignore list, but instead of adding multiple URLs to this list I want to configure just single URL to use the proxy.

**Additional context**

**Comment by npepinpe** [comment]
Hi @Aafi01, have you tried launching your client application with the parameters `-Dhttps.proxyHost=` and `-Dhttps.proxyPort=`? For example, if you have a proxy at `localhost:3909`, then you could start your client application with the properties `-Dhttps.proxyHost=localhost -Dhttps.proxyPort=3909` set. `grpc-java` looks at system properties as well to configure itself. I agree that ideally you could configure this via Zeebe, but perhaps this can help you for now.

**Comment by Aafi01** [comment]
Yeah I have tried setting these JVM parameters. Issue with this was that proxy requires authentication also, and there was no way to pass those credentials using these parameters. So I ended up using `-Djava.net.useSystemProxies=true` and enabling proxy at system level. This works.

Even if the proxy didn't require authentication and I used `-Dhttps.proxyHost=` and `-Dhttps.proxyPort=` to start the client, I still have to set the URLs which need to bypass proxy (which ideally I don't want to do).

**Comment by npepinpe** [comment]
Gotcha. I think there's also the proxy parameters for authentication (`-Dhttps.proxyUser=myuser` `-Dhttps.proxyPassword=mypass`), but if you have any more complex authentication I guess you're out of luck with standard Java parameters.

Regardless, I also agree it makes sense to allow configuring this in the client anyway. I was just hoping to unblock you in the meantime :+1:

**Comment by ClaraPaiva** [comment]
Hello,
Did this happen to work, for you? 
I have the same problem but it doesn't work for me...
